packages/datetime/src/dateInput.tsx

```diff
@@ -472,35 +480,38 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     };
 
     private registerPopoverBlurHandler = () => {
-        const node = ReactDOM.findDOMNode(this);
-        const tabbableElements = node.querySelectorAll("[tabindex]:not([tabindex='-1'])");
-        const numOfElements = tabbableElements.length;
-        if (numOfElements) {
-            const lastPopoverElement = tabbableElements[numOfElements - 1] as HTMLElement;
-            if (this.lastPopoverElement !== lastPopoverElement) {
-                this.unregisterPopoverBlurHandler();
-                this.lastPopoverElement = lastPopoverElement;
-                this.lastPopoverElement.addEventListener("blur", this.handlePopoverBlur);
+        if (this.contentRef) {
+            // Popover contents are well structured, but the selector will need
+            // to be updated if more focusable components are added in the future
+            const tabbableElements = this.contentRef.querySelectorAll("input, [tabindex]:not([tabindex='-1'])");
+            const numOfElements = tabbableElements.length;
+            if (numOfElements > 0) {
+                const lastPopoverElement = tabbableElements[numOfElements - 1] as HTMLElement;
+                if (this.lastPopoverElement !== lastPopoverElement) {
+                    this.unregisterPopoverBlurHandler();
+                    this.lastPopoverElement = lastPopoverElement;
+                    this.lastPopoverElement.addEventListener("blur", (this.handleClosePopover as any) as EventListener);
+                }
             }
         }
     };
 
     private unregisterPopoverBlurHandler = () => {
         if (this.lastPopoverElement) {
-            this.lastPopoverElement.removeEventListener("blur", this.handlePopoverBlur);
+            this.lastPopoverElement.removeEventListener("blur", (this.handleClosePopover as any) as EventListener);
```

Comment:
oh whoa well if the alternative is an `as any` then just do the `handlePopoverBlur`, but put it all on one line with a comment.

```
// blur DOM event listener (not React event)
private handlePopoverBlur = () => this.handleClosePopover();
```

Reply:
- [ ] use snippet above